### ISSUE TITLE
Port module is identified as invalid module declaration in 0.18.3

### ISSUE THREAD

**Issue by stil4m** [rtfeldman/node-test-runner#143]
When I migrated from the old elm-test version (0.18.2)  to the newer (0.18.3) by following the instructions [here](https://groups.google.com/forum/#!topic/elm-dev/7mmElgi4QM8) I ran into the following message:

```
$PWD/tests/Main.elm has an invalid module declaration. Check the first line of the file and make sure it has a valid module declaration there!
```

This was due to the fact that I left the `port` keyword in front of the `Main` module:

```
port module Main exposing (..)
```

Removing this fixed the issue. It may be nice to detect the left over `port` keyword and inform people that are migrating that they should remove this. The current error message is a bit misleading, because the declaration is actually valid and did not touch it in the migration process (before I removed the keyword).

**Comment by brunogirin** [comment]
I have the same problem on my CI server but not on my laptop which uses 0.18.3 as well.

The CI server is a Ubuntu 16.04 Docker container.
The laptop is Ubuntu 16.04 (no Docker).

**Comment by brunogirin** [comment]
Regarding the workaround suggested by @stil4m if you remove the `port` keyword, the module doesn't compile because of the `emit` port defined further down in the file. What other changes need to be done to that file? Those other changes should be documented too. In addition, running `elm test init` still creates the old `Main.elm` file so it should be updated to create a file that `elm-test` is happy with.

However, is all this made moot by the new `elm-test` not requiring `Main.elm` any more?

**Comment by n1k0** [comment]
As mentioned in the bug description, you need to follow the new setup instructions https://groups.google.com/forum/#!topic/elm-dev/7mmElgi4QM8

I'm having a hard time having this new setup working especially since `package.elm-lang.org` is down haha http://downforeveryoneorjustme.com/http://package.elm-lang.org

**Comment by stil4m** [comment]
@brunogirin In the upgrade path descriped in on elm-dev, you are instructed to remove the `emit` port definition, but not the `port` keyword.

**Comment by eeue56** [comment]
This should work. It's my fault. Fix incoming

**Comment by eeue56** [comment]
Unable to reproduce. @stil4m Can you create an failing example?

**Comment by stil4m** [comment]
@eeue56 I've just set up [this repo](https://github.com/stil4m/node-test-runner-issue-143) in the failing state. The [first commit](https://github.com/stil4m/node-test-runner-issue-143/commit/389b24788bb6ba7d4c999374f6d64f03bba4ff89) is the initialisation with elm-test `0.18.2`. The [second commit](https://github.com/stil4m/node-test-runner-issue-143/commit/6d26161ea7d10d0321965c24eb667956e245db14) is the migration.

After the second commit elm-test 0.18.3 gives the following output:

```
Success! Compiled 1 module.                                         
Successfully generated /dev/null
/Users/$USER/tmp/node-test-runner-issue-143/tests/Main.elm has an invalid module declaration. Check the first line of the file and make sure it has a valid module declaration there!
```

**Comment by eeue56** [comment]
Oh, I see. It's not my fault then. I'll still look at it anyway :) But for future reference, best practice is to _never_ import your `Main` file.

**Comment by stil4m** [comment]
I agree, but it is the `Main` file originally created by elm-test in the `tests` folder. We maybe should update the instructions that the users should just remove that file if it does not contain tests.

**Comment by rtfeldman** [comment]
Published fixed version in 0.18.4!
